### ISSUE TITLE
Number Extension Suggestion

### ISSUE THREAD

**Issue by Skyhigh173** [TurboWarp/extensions#301]
I am planning to add a extension which has following features : 
- BigInt (you can do +-*/% and everything that BigInt supports.)
- Decimal.js (custom sig fig, numbers > 1.79e308, maybe can replace BigInt too)
- other JS Math built-in func and const (pi, e, phi, omega, sqrt2, sqrt 0.5, max safe int etc)

**Comment by True-Fantom** [comment]
@Skyhigh173 Most of this is exist in Math.js, and I'm already doing new extension for Bigint https://turbowarp.org/editor?extension=https://true-fantom.github.io/tests/bigint.js

**Comment by Skyhigh173** [comment]
@True-Fantom I will do my own one with more features.

**Comment by True-Fantom** [comment]
@Skyhigh173 What I showed you is a prototype, I'm actively working on it. And if you are talking about Mathematical functions that are not in Math, they will also be added soon. So it's not necessary.

**Comment by True-Fantom** [comment]
@Skyhigh173 Can you please list all the functions that you lack so that I can add everything for sure.

**Comment by True-Fantom** [comment]
@Skyhigh173 One more question. Are there any other useful constants besides these?
![Скриншот 24-03-2023 165844](https://user-images.githubusercontent.com/116278271/227541835-64a61f2c-af3c-400e-8d2f-253ebcf88528.png)

**Comment by Skyhigh173** [comment]
@True-Fantom why not make it a menu? (You can check JS Math built-in const)

**Comment by True-Fantom** [comment]
@Skyhigh173 

> @True-Fantom You can check JS Math built-in const

I've already checked. Of the useful ones there are only `pi` and `e`, which is why I decided to ask you if you know any other constants (for example, I didn't know about the existence of omega at all)

> @True-Fantom why not make it a menu?

To do this, you would have to remove existing blocks from the palette, and it would also take up more space (And also the menu would not be very useful if I left special characters, and them greatly shorten the length of blocks)

**Comment by penta-quark-neutro** [comment]
Could you add support for complex numbers?
or the factorial operation x!

**Comment by True-Fantom** [comment]
@penta-quark-neutro Yes, I'm going to add a factorial in the next Math update, but this is the first time I've heard about complex numbers.. What do they do and what blocks should be added for them?

**Comment by penta-quark-neutro** [comment]
The complex numbers are the superior set to the real numbers that we use all the time (basically R∈C).
Currently I do not have my computer at hand, and there is a way to calculate them.
But basically a whole new interface is going to be created for these numbers since they are a composition (a+bi). Adding, subtracting, and multiplying them is easy, but it can be a bit more difficult with things like their logarithms and roots.
Basically the imaginary number i=root of -1, its simplest definition.

**Comment by Skyhigh173** [comment]
> The complex numbers are the superior set to the real numbers that we use all the time (basically R∈C).
> Currently I do not have my computer at hand, and there is a way to calculate them.
> But basically a whole new interface is going to be created for these numbers since they are a composition (a+bi). Adding, subtracting, and multiplying them is easy, but it can be a bit more difficult with things like their logarithms and roots.
> Basically the imaginary number i=root of -1, its simplest definition.
> 

I have already done a complex number extension
But it needs testing

**Comment by penta-quark-neutro** [comment]
Since I don't know how to create extensions, I did this to calculate complex numbers.
it is not practical since round reporters are better and those blocks are not practical.
I also could not define the complex factorial.
![Captura de pantalla (209)](https://github.com/TurboWarp/extensions/assets/134885702/df4a65c3-493e-4f74-988b-647f1ba008b3)

**Comment by SharkPool-SP** [comment]
Can this be closed?

**Comment by Brackets-Coder** [comment]
> Can this be closed?

I'd assume so, there's a few open PRs that do this
